### PR TITLE
Add Python 3 Support

### PR DESCRIPTION
**Pull request by AlexVerrico**
As Python 2 has now officially reached End Of Life ([https://www.python.org/doc/sunset-python-2/](https://www.python.org/doc/sunset-python-2/)), and Octoprint supports Python 3 as of [Release 1.4.0](https://octoprint.org/blog/2020/03/04/new-release-candidate-1.4.0/), all plugins are encouraged to add Python 3 compatibility, which is what I have done in this pull request.

A brief summary of the changes that I made:
- Added the plugin property `__plugin_pythoncompat__ = ">=2.7,<4"`
- Replaced ThreadPool.py with a new version that has support for both Python 2 and 3.
- Updated the plugin version to 0.1.4

This update has being tested and confirmed to work in both Python 2.7 and 3.7 using Octoprint 1.4.0 running on Octopi 0.17.0

Cheers,
Alex.

### CODE REVIEW

File: octoprint_filemanager/ThreadPool.py [hunk 1]
```diff
@@ -1,33 +1,51 @@
-# http://code.activestate.com/recipes/577187-python-thread-pool/
```
Comment (Salandora) [inline octoprint_filemanager/ThreadPool.py:1]: Can we readd the comment?

I really can't see any major rewrite that would justify removing the link to the original code.

File: octoprint_filemanager/ThreadPool.py [hunk 2]
```diff
@@ -1,33 +1,51 @@
...
     def add_task(self, func, *args, **kargs):
-        """Add a task to the queue"""
+        """ Add a task to the queue """
         self.tasks.put((func, args, kargs))
 
+    def map(self, func, args_list):
+        """ Add a list of tasks to the queue """
+        for args in args_list:
+            self.add_task(func, args)
+
```
Comment (Salandora) [inline octoprint_filemanager/ThreadPool.py:48]: Out of curiosity did you intend to use that function anywhere?
I couldn't find a reference to it.

**Comment by Salandora** [comment]
Hello @AlexVerrico,

thank you for the PR.

I made 2 little comments in the ThreadPool code.
I really would like to readd the comment so anyone who reads that file can see where that code is based of.

Also I couldn't find any reference to the map function but I wouldn't mind letting it there. 
Could come in handy at some point.

**Comment by AlexVerrico** [comment]
Hi.
This code actually came from somewhere else and then I modified it slightly. I'm not in front of my PC right now, bit when I am I will add the comment with the link to where I got the start of this code from. Cheers, Alex.

**Comment by AlexVerrico** [comment]
Hi @Salandora 
The basis of the new ThreadPool.py came from here: https://www.metachris.com/2016/04/python-threadpool/
Cheers,
Alex.

**Comment by Salandora** [comment]
Hey @AlexVerrico,

as I guessed your code was also based on the same code, I added the comments and will merge your PR now.

**Pull request merged**